EpiAware/docs/src/getting-started/explainers/censored-obs.jl

```diff
@@ -0,0 +1,321 @@
+	vline!([mean(true_dist)], ls = :dash, c= :black, lw = 3, lab = "")
+end
+
+# ╔═╡ f66d4b2e-ed66-423e-9cba-62bff712862b
+md"
+We've aggregated the data to unique combinations of `pwindow`, `swindow`, and `obs_time` and counted the number of occurrences of each `observed_delay` for each combination. This is the data we will use to fit our model.
+"
+
+# ╔═╡ 010ebe37-782b-4a35-bf5c-dca6dc0fee45
+md"
+## Fitting a naive model using Turing
+
+We'll start by fitting a naive model using Turing.
+"
+
+# ╔═╡ d9d14c48-8700-42b5-89b4-7fc51d0f577c
+@model function naive_model(N, y, n)
+	mu ~ Normal(1., 1.)
+	sigma ~ truncated(Normal(0.5, 1.0); lower= 0.0)
+	d = LogNormal(mu, sigma)
+	
+	for i in eachindex(y)
+		Turing.@addlogprob! n[i] * logpdf(d, y[i])
+	end
+end
+
+# ╔═╡ 8a7cd9ec-5640-4f5f-84c3-ae3f465ca68b
+md"
+Now lets instantiate this model with data
+"
+
+# ╔═╡ 028ade5c-17bd-4dfc-8433-23aaff02c181
+naive_mdl = naive_model(
+	size(delay_counts,1), 
+	delay_counts.observed_delay .+ 1e-6, # Add a small constant to avoid log(0)
+	delay_counts.n)
+
+# ╔═╡ 04b4eefb-f0f9-4887-8db0-7cbb7f3b169b
+md"
+and now let's fit the compiled model.
+"
+
+# ╔═╡ 21655344-d12b-4e47-a9a9-d06bd909f6ea
+naive_fit = sample(naive_mdl, NUTS(), MCMCThreads(), 500, 4)
+
+# ╔═╡ 3b89fe00-6aaf-4764-8b29-e71479f1e641
+summarize(naive_fit)
+
+# ╔═╡ 43eac8dd-8f1d-440e-b1e8-85db9e740651
+md"
+We see that the model has converged and the diagnostics look good. However, just from the model posterior summary we see that we might not be very happy with the fit. `mu` is smaller than the target $(meanlog) and `sigma` is larger than the target $(sdlog).
+
+"
+
+# ╔═╡ b2efafab-8849-4a7a-bb64-ac9ce126ca75
+md"
+## Fitting an improved model using primarycensoreddist
+
+We'll now fit an improved model using the `censored_pmf` function from the `EpiAware.EpiAwareUtils` submodule. This accounts for the primary and secondary censoring windows as well as the truncation.
+
+"
+
+# ╔═╡ ef40112b-f23e-4d4b-8a7d-3793b786f472
+@model function primarycensoreddist_model(N, y, y_upper, n, pwindow, D)
+	try
+		mu ~ Normal(1., 1.)
+		sigma ~ truncated(Normal(0.5, 0.5); lower= 0.1,)
+		d = LogNormal(mu, sigma)
+		log_pmf = censored_pmf(d; Δd = pwindow, D = D) .|> log
+	
+		for i in eachindex(y)
+			Turing.@addlogprob! n[i] * log_pmf[y[i] + 1] #0 obs is first element of array
+		end
+		return log_pmf
+	catch
+		Turing.@addlogprob! -Inf
+	end
+end
+
+# ╔═╡ b823d824-419d-41e9-9ac9-2c45ef190acf
+md"
+Lets instantiate this model with data
+"
+
+# ╔═╡ 93bca93a-5484-47fa-8424-7315eef15e37
+primarycensoreddist_mdl = primarycensoreddist_model(
+	size(delay_counts,1), 
+	delay_counts.observed_delay, # Add a small constant to avoid log(0)
+	delay_counts.observed_delay_upper, # Add a small constant to avoid log(0)
+	delay_counts.n,
+	delay_counts.pwindow[1],
+	delay_counts.obs_time[1]
+)
+
+# ╔═╡ 8f1d32fd-f54b-4f69-8c93-8f0786366cef
+# ╠═╡ disabled = true
+#=╠═╡
+benchmark_model(
```

Comment:
I guess this is all stuff that doesn't;t work super well? Can we include with a try catch or similar etc so that we can compare/use this to iterate?

Reply:
No, it works in the sense it catches the errors... Sorry this is a WIP draft (I may have hit the review button by mistake...)